source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1384,7 +1397,11 @@ recovering" error::
 
         # Mark the server Unknown
         unknown = new ServerDescription(type=Unknown, error=message, code=code, topologyVersion=response["topologyVersion"])
-        onServerDescriptionChanged(unknown)
+        with client.lock:
```

Comment:
It may make more sense to hold the lock throughout this entire method to ensure that the topology is not updated after the `isStaleError` check but before the `client.lock`.

Reply:
done